src/compile/format.ts

```diff
@@ -211,6 +206,9 @@ export function guideFormat(
 
   if (isFieldOrDatumDefForTimeFormat(fieldOrDatumDef)) {
     const timeUnit = isFieldDef(fieldOrDatumDef) ? normalizeTimeUnit(fieldOrDatumDef.timeUnit)?.unit : undefined;
+    if (timeUnit === undefined && config.customFormatTypes && config.timeFormatType) {
```

Comment:
This section is problematic. Without it, the test with the label: _returns undefined for a field if custom timeFormatType is in the config_ fails. However, with it, the test labelled _returns prefers timeUnit to timeFormatType_ fails.